tinygrad/renderer/cstyle.py

```diff
@@ -137,6 +137,7 @@ def ssa(prefix:str, u:Optional[UOp]=None):
         elif uop is UOps.ALU:
           # remove parens if ALU types are the same. TODO: can do more here
           if args in {BinaryOps.ADD,BinaryOps.MUL,BinaryOps.XOR}: operands = [strip_parens(r[v]) if v.arg == args else r[v]for v in src]
+          elif args is BinaryOps.MAX: operands = [self.render_cast(r[v],v.dtype) if v.op is UOps.CONST and v.dtype is not None else r[v] for v in src]
```

Comment:
you can cast the dtype here with cast(DType
we assert CONST always has a dtype.

Reply:
Sure, changed